rbd/rbd_nautilus.go

```diff
@@ -45,3 +47,64 @@ func GetImageNames(ioctx *rados.IOContext) ([]string, error) {
 	}
 	return names, nil
 }
+
+// GetPoolMetadata returns pool metadata string associated with the given key.
+//
+// Implements:
+//  int rbd_pool_metadata_get(rados_ioctx_t io_ctx, const char *key, char *value, size_t *val_len);
+func GetPoolMetadata(ioctx *rados.IOContext, key string) (string, error) {
```

Comment:
I wonder if it makes more sense to have this called like `rados.IOContext.RbdPoolMetadataGet()` as a method of `IOContext`. That removes the 1st argument of the function, and is more inline with other methods for types.

Reply:
Then I guess I need to move these to rados package?
inside rados/ioctx.go

Reply:
Ok, will wait for the comments from @phlogistonjohn before moving these.
And whatever decision we take for these should apply to namespace functions also in this [issue](https://github.com/ceph/go-ceph/issues/276)?

Reply:
I think if a function comes from librbd it belongs in the go-ceph rbd package. Much like CreateImage or OpenImage takes a  `*rados.IOContext`  I think its fairly natural for some rbd functions to take input from rados, but not the other way around. However, I think it would be good to have a new file for these functions.

Reply:
I anyway had to create a new file (rbd_nautilus.go) because these functions exist nautilius onwards.
Let me know if I need to rename it.

Reply:
I'd prefer a name that better reflects the contents, like metadata.go or something along those lines.